extensions/2.0/Vendor/EXT_materials_bump/README.md

```diff
@@ -0,0 +1,142 @@
+
+## Overview
+
+This extension adds one parameters to the metallic-roughness material: `bump`.
+
+`bump` allows users to perturb the surface normal via a bump map. The bump map is a grayscale texture, where the value of each texel represents the height of the surface at that texel.  The height is then used to perturb the normal direction.
+
+Benefits of bump maps are primarily these two:
+
+1. Bump maps are much more compact that normal maps and lead to smaller glTF files as well as less GPU memory requirements.  This is because the normal map has to store a full 3D vector per texel, while the bump map only needs to store a single scalar value per texel.  Thus bump maps require 3 times less data than normal maps for transmission and GPU storage.  Normal maps are also very sensitive to texture compressions methods, while bump maps are less sensitive.
+
+2. Bump maps are useful for capturing small details in combination with a normal map for large scale structures.  For example, when rendering human faces, there are usually two layers of surface pertubation detail, the unwrapped normal map is used for representing wrinkles, and then a repeating bump map is used for skin pore details.  This is also the case for many other materials such as wood, metal, and stone, where there is both macro and micro details.
+
+## Extending Materials
+
+The `KHR_materials_bump` extension can be specified in the `extensions` property of the material:
+
+```json
+{
+    "materials": [
+        {
+            "extensions": {
+                "KHR_materials_bump": {
+                    "bumpFactor": 0.5
+                }
+            }
+        }
+    ]
+}
+```
+
+Factor and texture are combined by multiplication to describe a single value.
+
+| |Type|Description|Required|
+|-|----|-----------|--------|
+| **bumpFactor** | `number` | The strength of the specular reflection. | No, default: `1.0`|
+| **bumpTexture** | [`textureInfo`](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#reference-textureinfo) | A texture that defines the strength of the specular reflection, stored in the alpha (`A`) channel. This will be multiplied by bumpFactor. | No |
+
+The `bump` parameters affect the normal used in `dielectric_brdf` of the glTF 2.0 metallic-roughness material.
+
+## Implementation
+
+*This section is non-normative.*
+
+The `bump` parameter is used to perturb the normal direction. The normal direction is perturbed by the value of the bump map at the texel corresponding to the fragment. The perturbed normal is then used in the `dielectric_brdf` function of the glTF 2.0 metallic-roughness material.
+
+### Blending Normal and Bump Perturbations
+
+In order to properly blend the bump map with a normal map, please follow the recommendation of https://blog.selfshadow.com/publications/blending-in-detail/.
+
+```glsl
+vec3 t = originalNormal.xyz * vec3( 2,  2, 2 ) + vec3( -1, -1, 0 );
+vec3 u = bumpNormal.xyz * vec3( -2, -2, 2 ) + vec3( 1,  1, -1 );
```

Comment:
The reference in Stephen Hill's blog uses normal maps for both the base and detail layers, as Unity does. We would need to make adjustments to this code, if our implementation will instead use grayscale bump maps.

Reply:
I think that as long as you derive the normal from the bump map, you can use the same blending equation.